instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/src/test/java/io/opentelemetry/instrumentation/kafkaclients/KafkaClientPropagationDisabledTest.java

```diff
@@ -0,0 +1,118 @@
+                    .hasKind(SpanKind.PRODUCER)
+                    .hasNoParent()
+                    .hasAttributesSatisfying(
+                        equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION, SHARED_TOPIC),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION_KIND, "topic"),
+                        satisfies(
+                            SemanticAttributes.MESSAGING_KAFKA_PARTITION,
+                            AbstractLongAssert::isNotNegative),
+                        satisfies(
+                            AttributeKey.longKey("messaging.kafka.message.offset"),
+                            AbstractLongAssert::isNotNegative));
+              });
+        });
+
+    awaitUntilConsumerIsReady();
+
+    ConsumerRecords<?, ?> records = consumer.poll(Duration.ofSeconds(5).toMillis());
+    assertThat(records.count()).isEqualTo(1);
+
+    // iterate over records to generate spans
+    for (ConsumerRecord<?, ?> ignored : records) {
+      testing.runWithSpan("processing", () -> {});
+    }
+
+    testing.waitForTraces(2);
+    testing.waitAndAssertTraces(
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> {
+                span.hasName(SHARED_TOPIC + " send")
+                    .hasKind(SpanKind.PRODUCER)
+                    .hasNoParent()
+                    .hasAttributesSatisfying(
+                        equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION, SHARED_TOPIC),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION_KIND, "topic"),
+                        satisfies(
+                            SemanticAttributes.MESSAGING_KAFKA_PARTITION,
+                            AbstractLongAssert::isNotNegative),
+                        satisfies(
+                            AttributeKey.longKey("messaging.kafka.message.offset"),
+                            AbstractLongAssert::isNotNegative));
+              });
+        },
+        trace -> {
+          trace.hasSpansSatisfyingExactly(
+              span -> {
+                span.hasName(SHARED_TOPIC + " process")
+                    .hasKind(SpanKind.CONSUMER)
+                    .hasLinks(Collections.emptyList())
+                    .hasAttributesSatisfying(
+                        equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION, SHARED_TOPIC),
+                        equalTo(SemanticAttributes.MESSAGING_DESTINATION_KIND, "topic"),
+                        equalTo(SemanticAttributes.MESSAGING_OPERATION, "process"),
+                        equalTo(
+                            SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES,
+                            message.getBytes(StandardCharsets.UTF_8).length),
```

Comment:
The original code for this part is `"$SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES" Long`. I am not sure what was the intention, so changed to the message's byte size. 
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/kafka/kafka-clients/kafka-clients-0.11/javaagent/src/test/groovy/io/opentelemetry/instrumentation/kafkaclients/KafkaClientDefaultTest.groovy#L111

Reply:
I think we just wanted to make sure that the payload size gets captured at all.